next-jdbc/src/ragtime/next_jdbc.clj

```diff
@@ -60,14 +60,28 @@
                        (str table_schem "." table_name)
                        table_name)))
 
-(defn- table-exists? [datasource ^String table-name]
+(defn- table-exists-via-metadata-scan? [datasource ^String table-name]
   (let [{:keys [tables quote]} (get-db-metadata datasource)
         table-name-unquoted    (unquote-table-name quote table-name)]
     (some (partial metadata-matches-table? table-name-unquoted)
           tables)))
 
-(defn- ensure-migrations-table-exists [datasource migrations-table]
-  (when-not (table-exists? datasource migrations-table)
+(defn- table-exists-via-provided-sql? [datasource ^String migrations-table-exists-sql]
+  (let [row-count (count (jdbc/execute! datasource [migrations-table-exists-sql]))]
+    (case row-count
+      0 false
+      1 true
+      (throw (ex-info "Provided SQL doesn't directly answer if a migration table exists or not"
```

Comment:
Not sure if you like the approach with throwing an exception here. Or should we simply accept any `count >= 1` as "table exists"?

Reply:
Yep, I think any count greater than 1 should be true.